tests/test_reactive.py

```diff
@@ -0,0 +1,55 @@
+from textwrap import dedent
+
+from reacttrs import reactive
+
+
+class Foo:
+
+    name = reactive("Paul")
+    age = reactive(33)
+    birth = reactive(1990)
+
+    def watch_name(self, old, new):
+        print(f"{old=}, {new=}")
+
+    def validate_name(self, name):
+        if name == "John":
+            print("Hey John!")
+        return name
+
+    def compute_age(self) -> int:
+        age = 2023 - self.birth
+        print(f"{age=}")
+        return age
+
+
+def test(capfd):
+    foo = Foo()
+
+    foo.name = "John"
+    out, err = capfd.readouterr()
+    assert out == dedent("""\
+        old='Paul', new='Paul'
```

Comment:
The `watch_` method is called a first time with the same value.
I'm not sure it's the case in Textual @willmcgugan?

Reply:
It's not called first time, unless `init=True` IIRC

Reply:
Indeed, thanks. Any reason for having `init=True` as the default value?

Reply:
Generally in a Textual app, you want to respond to changes to the state. But there isn’t often a need to do anything when you first set the state.

It’s also problematic if your watcher makes assumptions about other widgets, that may not yet have updated their state. So init=False was the safer default.

Reply:
But it's the [opposite](https://github.com/Textualize/textual/blob/10e61987e36a45b8a174d179f86e2280e66cfb94/src/textual/reactive.py#L313), right?

Reply:
Ah, you’re right. Don’t recall why we changed it. Suspect it more aligned with user expectations.